src/fenwick.jl

```diff
@@ -0,0 +1,37 @@
+mutable struct FenwickTree{T}
+    BIT::Vector{T}
+    n::Integer
```

Comment:
For speed this should either be a concrete type (e.g. `Int`) or a type param.
I am leaning towards `Int`, there is afact no reason someone would want to use e.g. a `BigInt`.

Reply:
True

Reply:
Doesn't seem to have been addressed